docs/sts-migrate-to-private-bucket.md

```diff
@@ -0,0 +1,491 @@
+
+6) AWS policies required by the agent running that steps:
+
+```JSON
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "Stmt1678336730279",
+      "Action": [
+        "iam:CreateOpenIDConnectProvider",
+        "iam:DeleteOpenIDConnectProvider",
+        "iam:GetRole",
+        "iam:ListOpenIDConnectProviderTags",
+        "iam:ListOpenIDConnectProviders",
+        "iam:ListRoles",
+        "iam:UpdateAssumeRolePolicy"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    },
+    {
+      "Sid": "Stmt1678336953197",
+      "Action": [
+        "s3:GetObject",
+        "s3:ListBucket",
+        "s3:PutObject"
+      ],
+      "Effect": "Allow",
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+
+## Validate and Backup <a name="validate-backup"></a>
+
+### Validate tokens <a name="validate-token"></a>
+
+This section is to make sure everything is working correctly in your existing environment.
+
+The steps described below will use the credentials provided to the `machine-api` component, trying to assume the role using `aws-cli`.
+
+It's expected that the existing token will be able to authenticate in AWS. Otherwise, you must abort the operations and do not try to run any step described in the next sections
+
+```bash
+## test existing token
+# Get Token path from AWS credentials mounted to pod
+TOKEN_PATH=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^web_identity_token_file" |\
+    awk '{print$3}')
+
+# Get Controler's pod
+CAPI_POD=$(oc get pods -n openshift-machine-api \
+    -l api=clusterapi \
+    -o jsonpath='{.items[*].metadata.name}')
+
+# Extract tokens from the pod
+TOKEN=$(oc exec -n openshift-machine-api ${CAPI_POD} \
+    -c machine-controller -- cat ${TOKEN_PATH})
+
+echo $TOKEN | awk -F. '{ print $2 }' | base64 -d 2>/dev/null | jq .iss
+
+IAM_ROLE=$(oc get secrets aws-cloud-credentials \
+    -n openshift-machine-api \
+    -o jsonpath='{.data.credentials}' |\
+    base64 -d |\
+    grep "^role_arn" |\
+    awk '{print$3}')
+
+echo $IAM_ROLE
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}"
+```
+
+### Backup existing state <a name="backup"></a>
+
+- Get Objects and existing authentication response
+
+```bash
+export BACKUP_PATH=$PWD/current-cluster
+mkdir $BACKUP_PATH
+
+oc get authentication -o yaml |tee -a $BACKUP_PATH/authentication.yaml
+
+aws sts assume-role-with-web-identity \
+    --role-arn "${IAM_ROLE}" \
+    --role-session-name "my-session" \
+    --web-identity-token "${TOKEN}" \
+    | jq -r '.Credentials=""' \
+    | tee ${BACKUP_PATH}/identities.json
+```
+
+- Save the current IAM Roles
+
+```bash
+aws iam list-roles \
+    | jq -r  ".Roles[] | select(.RoleName | startswith(\"${CLUSTER_NAME}-openshift\"))" \
+    | tee ${BACKUP_PATH}/iam-roles.json
```

Comment:
If there's application role created following https://docs.openshift.com/rosa/authentication/assuming-an-aws-iam-role-for-a-service-account.html, then the role name doesn't follow the style of ${CLUSTER_NAME}-openshift.
So better to add a note that this kind of case is NOT supported by the procedure.

Reply:
Done https://github.com/openshift/cloud-credential-operator/pull/515/commits/c8f24f602088c33f38f5cc1fd03430200eb0cc0b